### ISSUE TITLE
Timeout

### ISSUE THREAD

**Issue by trazola** [frankie567/httpx-oauth#272]
Hi,
I get a "timeout" error when using the Oauth2 client for Facebook.
Is it possible to set the timeout parameter?

**Comment by frankie567** [comment]
Hi @trazola!

Could you give more details about the timeout you are getting? HTTPX has several kinds of timeout exceptions, so it would be worth to know which. Ideally, provide the stacktrace.

**Comment by frankie567** [comment]
Could you confirm that you have access to the Facebook endpoint from your machine. You can try this cURL command for example:

```sh
curl -XPOST "https://graph.facebook.com/v5.0/oauth/access_token"
```

**Comment by trazola** [comment]
I attach screenshots :)
![Screenshot from 2022-05-25 22-50-38](https://user-images.githubusercontent.com/81198593/170365150-65059c35-3e09-4b5f-8b4e-baa9a9b4cc3c.png)

![Screenshot from 2022-05-25 22-51-34](https://user-images.githubusercontent.com/81198593/170365241-26109a7b-7eae-4ec4-af8c-8365d7fec434.png)

**Comment by trazola** [comment]
The solution that helped me in another project where I was using HTTPX  was to set the timeout value to a higher value.
httpx.AsyncClient(timeout=30.0)

**Comment by frankie567** [comment]
I see, your connection time is surprisingly high, and HTTPX has a default timeout of 5 seconds. Are you behind a VPN or something?

I'll think about a way to tune HTTPX settings directly.

**Comment by frankie567** [comment]
I've released a new version `v0.7.0`, with the ability to customize the HTTPX client settings. You can see how in the documentation: https://frankie567.github.io/httpx-oauth/oauth2/#customize-httpx-client

**Comment by trazola** [comment]
Thank you for your help :).
